### PR TITLE
Make typename always include the module part II (THIS TIME, IT'S PERSONAL)

### PR DESCRIPTION
**Pull request by beOn**
Addresses #88. All tests pass after rebasing onto master.

### CODE REVIEW

File: test/internal.jl [hunk 1]
```diff
@@ -78,3 +78,7 @@ end
 @test JLD2.size_size(256) === 2
 @test JLD2.size_size(65536) === 4
 @test JLD2.size_size(UInt64(4294967296)) === 8
+
+# test with one of its own types
+println()
```
Comment (ararslan) [inline test/internal.jl]: Why is this necessary?

Comment (beOn) [reply test/internal.jl]: Um, obviously as a probe of your vigilance, duh. And congratulations - you passed.

Comment (beOn) [reply test/internal.jl]: (removed...)

Comment (beOn) [reply test/internal.jl]: (and apologies for the slops...)

Comment (ararslan) [reply test/internal.jl]: lol, no worries
